src/app/asset-page/artstor-viewer/artstor-viewer.component.ts

```diff
@@ -423,6 +426,18 @@ export class ArtstorViewerComponent implements OnInit, OnDestroy {
         })
 
         this.osdViewer.addHandler('zoom', (value: any) => {
+            if (this.autoZoomFinished && value.eventSource.id !== this._storage.getSession('loggedZoomAsset')) {
+                this._log.log({
+                    eventType: 'artstor_zoom',
+                    additional_fields: {
+                        item_id: this._assetId,
+                        fullscreen: this.isFullscreen
+                    }
+                })
+                this._storage.setSession('loggedZoomAsset', value.eventSource.id)
```

Comment:
While it might have been able possible to use a local variable to manage whether the log has fired, this is a really robust way to handle it! In the future, this is a great place for a comment to explain the logic, but nicely done.